### PR TITLE
add namespaces info for logPrefix

### PR DESCRIPTION
**Pull request by zhipengzuo**
**What type of PR is this?**
/kind cleanup
**What this PR does / why we need it**:
Different pod may have same name but with different namespace so we need to add namespace info .
**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
People can see the more effective image log.
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by zhipengzuo** [comment]
/assign @sjenning
PTAL,thanks.

File: pkg/kubelet/images/image_manager.go [hunk 1]
```diff
@@ -86,7 +86,7 @@ func (m *imageManager) logIt(ref *v1.ObjectReference, eventtype, event, prefix,
 // EnsureImageExists pulls the image for the specified pod and container, and returns
 // (imageRef, error message, error).
 func (m *imageManager) EnsureImageExists(pod *v1.Pod, container *v1.Container, pullSecrets []v1.Secret, podSandboxConfig *runtimeapi.PodSandboxConfig) (string, string, error) {
-	logPrefix := fmt.Sprintf("%s/%s", pod.Name, container.Image)
+	logPrefix := fmt.Sprintf("%s/%s/%s", pod.Namespace, pod.Name, container.Image)
 	ref, err := kubecontainer.GenerateContainerRef(pod, container)
 	if err != nil {
 		klog.Errorf("Couldn't make a ref to pod %v, container %v: '%v'", pod.Name, container.Name, err)
```
Comment (SergeyKanzhelev) [inline pkg/kubelet/images/image_manager.go:92]: what about this error message? Should it also include the `pod.Namespace`?

**Review by SergeyKanzhelev — COMMENTED** [review]
There are more places where this can be improved. Just a few first files in a search:

```
pkg/controller/daemon/daemon_controller.go:
   485: 		klog.V(4).Infof("Pod %s added.", pod.Name)
   499: 	klog.V(4).Infof("Orphan Pod %s added.", pod.Name)
   602: 	klog.V(4).Infof("Pod %s deleted.", pod.Name)

pkg/controller/daemon/update.go:
  65: 		klog.V(4).Infof("Marking pod %s/%s for deletion", ds.Name, pod.Name)
  75: 		klog.V(4).Infof("Marking pod %s/%s for deletion", ds.Name, pod.Name)

pkg/controller/deployment/deployment_controller.go:
  357: 	klog.V(4).Infof("Pod %s deleted.", pod.Name)
  425: 		klog.V(4).Infof("Cannot get replicaset %q for pod %q: %v", controllerRef.Name, pod.Name, err)

pkg/controller/disruption/disruption.go:
  383: 	klog.V(4).Infof("addPod called on pod %q", pod.Name)
  386: 		klog.V(4).Infof("No matching pdb for pod %q", pod.Name)
  389: 	klog.V(4).Infof("addPod %q -> PDB %q", pod.Name, pdb.Name)
  395: 	klog.V(4).Infof("updatePod called on pod %q", pod.Name)
  398: 		klog.V(4).Infof("No matching pdb for pod %q", pod.Name)
  401: 	klog.V(4).Infof("updatePod %q -> PDB %q", pod.Name, pdb.Name)
  424: 	klog.V(4).Infof("deletePod called on pod %q", pod.Name)
  427: 		klog.V(4).Infof("No matching pdb for pod %q", pod.Name)
  430: 	klog.V(4).Infof("deletePod %q -> PDB %q", pod.Name, pdb.Name)
  458: 		klog.V(4).Infof("No PodDisruptionBudgets found for pod %v, PodDisruptionBudget controller will avoid syncing.", pod.Name)
  651: 			err = fmt.Errorf("found no controller ref for pod %q", pod.Name)
  676: 			err = fmt.Errorf("found no controllers for pod %q", pod.Name)

pkg/controller/endpoint/endpoints_controller.go:
  429: 			klog.V(2).Infof("failed to find endpoint for service:%v with ClusterIP:%v on pod:%v with error:%v", service.Name, service.Spec.ClusterIP, pod.Name, err)

pkg/controller/replicaset/replica_set.go:
  379: 		klog.V(4).Infof("Pod %s created: %#v.", pod.Name, pod)
  393: 	klog.V(4).Infof("Orphan Pod %s created: %#v.", pod.Name, pod)

pkg/controller/statefulset/stateful_set.go:
  177: 		klog.V(4).Infof("Pod %s created, labels: %+v", pod.Name, pod.Labels)
  188: 	klog.V(4).Infof("Orphan Pod %s created, labels: %+v", pod.Name, pod.Labels)

pkg/controller/util/node/controller_utils.go:
   82: 		recorder.Eventf(pod, v1.EventTypeNormal, "NodeControllerEviction", "Marking for deletion Pod %s from Node %s", pod.Name, nodeName)
  140: 				klog.V(2).Infof("Updating ready status of pod %v to false", pod.Name)

pkg/kubelet/cm/cpumanager/cpu_manager.go:
  360: 			klog.Warningf("[cpumanager] reconcileState: skipping pod; status not found (pod: %s)", pod.Name)
  370: 				klog.Warningf("[cpumanager] reconcileState: skipping container; ID not found in pod status (pod: %s, container: %s, error: %v)", pod.Name, container.Name, err)
  377: 				klog.Warningf("[cpumanager] reconcileState: skipping container; container status not found in pod status (pod: %s, container: %s, error: %v)", pod.Name, container.Name, err)
  384: 				klog.Warningf("[cpumanager] reconcileState: skipping container; container still in the waiting state (pod: %s, container: %s, error: %v)", pod.Name, container.Name, err)
  397: 					klog.Warningf("[cpumanager] reconcileState: ignoring terminated container (pod: %s, container id: %s)", pod.Name, containerID)
  410: 				klog.Infof("[cpumanager] reconcileState: skipping container; assigned cpuset is empty (pod: %s, container: %s)", pod.Name, container.Name)
  415: 			klog.V(4).Infof("[cpumanager] reconcileState: updating container (pod: %s, container: %s, container id: %s, cpuset: \"%v\")", pod.Name, container.Name, containerID, cset)
  418: 				klog.Errorf("[cpumanager] reconcileState: failed to update container (pod: %s, container: %s, container id: %s, cpuset: \"%v\", error: %v)", pod.Name, container.Name, containerID, cset, err)

pkg/kubelet/cm/cpumanager/fake_cpu_manager.go:
  44: 	klog.Infof("[fake cpumanager] Allocate (pod: %s, container: %s", pod.Name, container.Name)
  49: 	klog.Infof("[fake cpumanager] AddContainer (pod: %s, container: %s, container id: %s)", pod.Name, container.Name, containerID)

pkg/kubelet/cm/cpumanager/policy_static.go:
  220: 		klog.Infof("[cpumanager] static policy: Allocate (pod: %s, container: %s)", pod.Name, container.Name)
  225: 			klog.Infof("[cpumanager] static policy: container already present in state, skipping (pod: %s, container: %s)", pod.Name, container.Name)
  236: 			klog.Errorf("[cpumanager] unable to allocate %d CPUs (pod: %s, container: %s, error: %v)", numCPUs, pod.Name, container.Name, err)
  350: 	klog.Infof("[cpumanager] TopologyHints generated for pod '%v', container '%v': %v", pod.Name, container.Name, cpuHints)

pkg/kubelet/cm/topologymanager/topology_manager.go:
  180: 		klog.Infof("[topologymanager] TopologyHints for pod '%v', container '%v': %v", pod.Name, container.Name, hints)

pkg/kubelet/config/common.go:
  69: 		klog.V(5).Infof("Generated UID %q pod %q from %s", pod.UID, pod.Name, source)
  73: 	klog.V(5).Infof("Generated Name %q for UID %q from URL %s", pod.Name, pod.UID, source)
  78: 	klog.V(5).Infof("Using namespace %q for pod %q from %s", pod.Namespace, pod.Name, source)

pkg/kubelet/config/config.go:
  355: 			klog.Warningf("Pod[%d] (%s) from %s failed validation due to duplicate pod name %q, ignoring", i+1, format.Pod(pod), source, pod.Name)
  356: 			recorder.Eventf(pod, v1.EventTypeWarning, events.FailedValidation, "Error validating pod %s from %s due to duplicate pod name %q, ignoring", format.Pod(pod), source, pod.Name)

pkg/kubelet/container/helpers.go:
  284: 	return fmt.Sprintf("%s_%s(%s)", pod.Name, pod.Namespace, pod.ID)

pkg/kubelet/container/ref.go:
  84: 	return "", fmt.Errorf("container %q not found in pod %s/%s", container.Name, pod.Namespace, pod.Name)

pkg/kubelet/images/image_gc_manager.go:
  231: 			klog.V(5).Infof("Pod %s/%s, container %s uses image %s(%s)", pod.Namespace, pod.Name, container.Name, container.Image, container.ImageID)

pkg/kubelet/images/image_manager.go:
  89: 	logPrefix := fmt.Sprintf("%s/%s", pod.Name, container.Image)
  92: 		klog.Errorf("Couldn't make a ref to pod %v, container %v: '%v'", pod.Name, container.Name, err)

pkg/kubelet/prober/prober.go:
  166: 		klog.V(4).Infof("Exec-Probe Pod: %v, Container: %v, Command: %v", pod.Name, container.Name, p.Exec.Command)
```

Inconsistent ordering of a namespace and name: 

```
pkg/kubelet/kuberuntime/helpers.go:
  161: 	return fmt.Sprintf("%s_%s_%s_%s_%s", pod.Name, pod.Namespace, string(pod.UID), container.Name, hash)

pkg/kubelet/pleg/generic.go:
  252: 				klog.V(4).Infof("PLEG: Ignoring events for pod %s/%s: %v", pod.Name, pod.Namespace, err)
  288: 					klog.V(5).Infof("PLEG: pod %s/%s failed reinspection: %v", pod.Name, pod.Namespace, err)
  386: 	klog.V(4).Infof("PLEG: Write status for %s/%s: %#v (err: %v)", pod.Name, pod.Namespace, status, err)
```

**Comment by zhipengzuo** [comment]
anyone can help me to find why the "pull-kubernetes-e2e-kind " test failed?  in the log some volume not ready but i don't think that's related to this pr

**Comment by zhipengzuo** [comment]
@matthyx hi,could you help me merge this pr? thanks.

**Comment by matthyx** [comment]
Let's merge this until the KEP is implemented: https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/1602-structured-logging
/lgtm
/assign @dashpole 
/priority backlog

**Comment by zhipengzuo** [comment]
anyone can help to merge this pr? thanks.

**Pull request merged**